lib/JSON/Validator/OpenAPI/Mojolicious.pm

```diff
@@ -89,7 +89,11 @@ sub validate_request {
       $value = $self->_coerce_by_collection_format($value, $p);
     }
 
-    ($exists, $value) = (1, $p->{default}) if !$exists and exists $p->{default};
+    {
+      local $p->{default} //= $p->{schema}{default}
+        if $p->{schema} && exists $p->{schema}{default}; # v3
+      ($exists, $value) = (1, $p->{default}) if !$exists and exists $p->{default};
```

Comment:
Can't you do this instead?

```
($exists, $value) = (1, $p->{schema}{default}) if !$exists and exists $p->{schema}{default};
($exists, $value) = (1, $p->{default}) if !$exists and exists $p->{default};
```

Reply:
I think we should avoid an autovivification calling exists() at the first line. With v2, a key "schema" would be added in $p with an empty hash inside.
Some tests actually are broken with this line.

This could be an alternative :

```
    ($exists, $value) = (1, $p->{schema}{default})
      if !$exists and $p->{schema} and exists $p->{schema}{default};
    ($exists, $value) = (1, $p->{default})
      if !$exists and exists $p->{default};
```